marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java

```diff
@@ -246,12 +247,10 @@ public void exportRows(JsonNode queryDocument, Long limit, OutputStream out) {
         List<SearchQuery.SortOrder> sortOrder = searchQuery.getSortOrder().orElse(new ArrayList<>());
         ArrayNode sortOrderNode = sortOrderToArrayNode(sortOrder);
 
-
-        JsonNode opticPlanNode = EntitySearchService.on(finalDatabaseClient).getOpticPlan(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream());
-        StringHandle stringHandle = new StringHandle(opticPlanNode.toString());
-        RowManager rowManager = finalDatabaseClient.newRowManager();
-        try (ReaderHandle readerHandle = new ReaderHandle()) {
-            rowManager.resultDoc(rowManager.newRawPlanDefinition(stringHandle), readerHandle.withMimetype(CSV_CONTENT_TYPE));
+        EntitySearchService entitySearchService = EntitySearchService.on(finalDatabaseClient);
+        // Exporting directly from Data Service to avoid bug https://bugtrack.marklogic.com/55338 related to namespaced path range indexes
+        Reader export = entitySearchService.exportSearchAsCSV(structuredQuery, searchText, queryOptions, entityTypeId, entityTypeId, limit, sortOrderNode, columns.stream());
+        try (ReaderHandle readerHandle = new ReaderHandle(export).withMimetype(CSV_CONTENT_TYPE)) {
```

Comment:
@akshaysonvane I know this code existed already, but I don't think it's necessary to call withMimetype. I looked at the underlying code, and I don't think it cares. It's using a Java Client class called Utilities, which does the same thing as the Spring class FileCopyUtils. I think to avoid confusion about the necessity of withMimetype (I'm 99% sure it's not needed) and for consistency (FileCopyUtils is a favorite class of mine since it solves the terrible user experience in Java of copying data around), it'd be better to just do:

FileCopyUtils.copy(export, new OutputStreamWriter(out)); 

(which will still need to be in a try/catch block)

Reply:
Yes, that's right. I looked and it seems withMimetype was required earlier when we were sending the optic plan to the RowManager as the default was JSON. Now that we are generating the CSV in the DS itself FileCopyUtils can be used for copying.

Reply:
Updated to use FileCopyUtils.copy.